### PR TITLE
1 MVP

### PR DESCRIPTION
**Pull request by nleanba**
See #1 for explanation.

This PR simplifies the external API and adds documentation (format is compatible with deno.land, so it should generate nice documentation there)

This PR does not concern itself with internal cleanup, pre-existing mess was left as-is if it doesn't touch the module user.

### CODE REVIEW

**Comment by retog** [comment]
The build fails because of missin web dir.

**Comment by nleanba** [comment]
Thanks for catching that, but also I didn't add that to the dockerfile. must've not built before either

**Comment by retog** [comment]
What's the rationale for moving the `Config` and `Job` interface to mod.ts?

**Review by retog — COMMENTED** [review]
I'm wondering if having the paths in `deno.json` and gettin rid of this file wouldn't be the better solution.

File: src/GitRepository.ts [hunk 1]
```diff
@@ -1,46 +1,83 @@
...
 
-export default class GitRepository {
-  /** @param repoUrl url of the repository */
   constructor(
-    protected repoUrl: string,
-    protected branch: string,
-    protected token: string | undefined,
-    public workDir: string,
+    uri: string,
+    branch: string,
```
Comment (retog) [inline src/GitRepository.ts:42]: why remove the modifier and then having to assign it to a field?

Comment (retog) [reply src/GitRepository.ts:42]: Okay, I think that's for them to be readonly, makes sense.

File: README.md [hunk 2]
```diff
@@ -19,6 +20,32 @@ This webserver also exposes the follwing paths:
...
-See the [example folder](example/)
+Documentation is available on [deno.land](https://deno.land/x/ghact?doc).
+
+### Example Usage
+
+main.ts:
+
+```ts
+import { GHAct, type Config } from "."
+const config: Config = { ... };
```
Comment (retog) [inline README.md:33]: This allows the config in the server and the worker to be different, as this usually shouldn't be the case, I suggest havin a `config.ts` module used by both.

**Review by retog — COMMENTED** [review]
All in all very nice improvement. Uncertain about the interfaces in `mod.ts`, it seems to go against the purpose of that file.

File: src/GitRepository.ts (hunk 1 above)
Comment (nleanba) [reply src/GitRepository.ts:42]: `// non-private members are declared this way and not via the constructor to enable documentation.`

**Comment by nleanba** [comment]
my rationale in moving the interfaces to `mod.ts` was that they are project-wide interfaces and this keeps the files with actual code much shorter, making it easier (as ghact developers) to find where code needs to be edited.

The purpose of `mod.ts` is to serve as the single file that needs to be imported by consumers of ghact. The only reason most of the code (logic) is _not_ in that file is to make the source easier to navigate, but there is no reason why `mod.ts` should not contain declarations itself.

The reasoning behind keeping a `deps.ts` file (I’m assuming thats what your comment is about) is to allow for caching for docker-builds, as far as I can tell the `deno.json` file doesn't allow for that.

**Review by retog — APPROVED** [review]
Good work!

**Pull request merged**